guppylang/compiler/expr_compiler.py

```diff
@@ -167,6 +170,10 @@ def visit_Constant(self, node: ast.Constant) -> Wire:
         raise InternalGuppyError("Unsupported constant expression in compiler")
 
     def visit_PlaceNode(self, node: PlaceNode) -> Wire:
+        if subscript := contains_subscript(node.place):
+            if subscript.item.id not in self.dfg.locals:
+                self.dfg[subscript.item] = self.visit(subscript.item_expr)
```

Comment:
just a side-thought but it is a bit confusing to read that you check for existence of `item.id`in `dfg.locals` but then just assign to `dfg[item]`. Does `dfg` need a`__contains__` ?